lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -4,79 +4,64 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React from 'react';
-import {mount, shallow} from 'enzyme';
-import 'jest-enzyme';
-
-import { getMountedComponent } from '../../../__testutils__/enzymeUtils';
+import React, { RefAttributes } from 'react';
 import NxModal, { Props } from '../NxModal';
-import NxTooltip from '../../NxTooltip/NxTooltip';
-import NxButton from '../../NxButton/NxButton';
-import { Tooltip } from '@material-ui/core';
-import { runTimers } from '../../../__testutils__/rtlUtils';
+import { rtlRender, rtlRenderElement, runTimers, userEvent } from '../../../__testutils__/rtlUtils';
+import { render, within } from '@testing-library/react';
 
 describe('NxModal', function() {
-  const dummyCloseHandler = jest.fn();
   const minimalProps: Props = {
     children: 'A message to show in a modal',
-    onClose: dummyCloseHandler
+    onClose: jest.fn()
   };
 
-  const getMounted = getMountedComponent<Props>(NxModal, minimalProps),
-      getModal = (props?: Partial<Props>) => getMounted(props).find('dialog');
+  type PropsWithRef = Props & RefAttributes<HTMLDialogElement>;
+  const quickRender = rtlRender<PropsWithRef>(NxModal, minimalProps),
+      renderEl = rtlRenderElement<PropsWithRef>(NxModal, minimalProps);
 
-  it('renders an nx-modal-backdrop <dialog> containing an nx-modal <div>', function () {
-    const nxModal = getModal();
+  it('renders a top-level element with role="dialog"', function () {
+    const view = quickRender(),
+        dialog = view.getByRole('dialog');
 
-    expect(nxModal).toMatchSelector('dialog.nx-modal-backdrop');
-    expect(nxModal.children()).toMatchSelector('div.nx-modal');
+    expect(dialog).toBeInTheDocument();
+    expect(dialog).toBe(view.container.firstElementChild);
   });
 
   it('renders children nodes within the modal', function() {
-    const nxModal = getModal({ children: <div className="bar"/> });
+    const view = quickRender({ children: <div className="bar" data-testid="test-div" /> }),
```

Comment:
```suggestion
    const view = quickRender({ children: <div data-testid="test-div" /> }),
```
Probably fine without the className included